### PR TITLE
Event stream API

### PR DESCRIPTION
**Pull request by darioAnongba**
Event stream API:
- Adds Delete, Update and Get use cases and DA functions
- Adds complete API endpoint + swagger
- Fixes swagger issues
- Adds SDK Event stream client
- Integration tests
- Fixes creation logic

### CODE REVIEW

File: src/api/business/use-cases/event_streams/delete.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	if err != nil {
+		return errors.FromError(err).ExtendComponent(deleteEventStreamComponent)
+	}
+
+	err = uc.db.Delete(ctx, uuid, userInfo.AllowedTenants, userInfo.Username)
+	if err != nil {
+		return errors.FromError(err).ExtendComponent(deleteEventStreamComponent)
+	}
+
+	logger.Info("event stream was deleted successfully")
```
Comment (julien-marchand) [inline src/api/business/use-cases/event_streams/delete.go:42]: Can we add event stream uuid and name here?

File: src/api/business/use-cases/event_streams/get.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+func (uc *getUseCase) Execute(ctx context.Context, uuid string, userInfo *multitenancy.UserInfo) (*entities.EventStream, error) {
+	ctx = log.WithFields(ctx, log.Field("event_stream_uuid", uuid))
+	logger := uc.logger.WithContext(ctx)
+
+	e, err := uc.db.FindOneByUUID(ctx, uuid, userInfo.AllowedTenants, userInfo.Username)
+	if err != nil {
+		return nil, errors.FromError(err).ExtendComponent(getEventStreamComponent)
+	}
+
+	logger.Debug("event stream found successfully")
```
Comment (julien-marchand) [inline src/api/business/use-cases/event_streams/get.go:37]: Can we add event stream uuid and name here?

File: src/api/business/use-cases/event_streams/create.go [hunk 3]
```diff
@@ -14,39 +15,55 @@ import (
...
+		errMsg := "event stream already exists"
 		logger.Error(errMsg)
 		return nil, errors.AlreadyExistsError(errMsg).ExtendComponent(createEventStreamComponent)
 	}
 
+	eventStream.TenantID = userInfo.TenantID
+	eventStream.OwnerID = userInfo.Username
 	e, err := uc.db.Insert(ctx, eventStream)
 	if err != nil {
 		return nil, errors.FromError(err).ExtendComponent(createEventStreamComponent)
```
Comment (julien-marchand) [inline src/api/business/use-cases/event_streams/create.go:69]: Can we add event stream uuid in the logger below?

File: src/api/business/use-cases/event_streams/search.go [hunk 4]
```diff
@@ -26,11 +26,11 @@ func NewSearchUseCase(db store.EventStreamAgent) usecases.SearchEventStreamsUseC
 }
 
 func (uc *searchUseCase) Execute(ctx context.Context, filters *entities.EventStreamFilters, userInfo *multitenancy.UserInfo) ([]*entities.EventStream, error) {
-	accs, err := uc.db.Search(ctx, filters, userInfo.AllowedTenants, userInfo.Username)
+	es, err := uc.db.Search(ctx, filters, userInfo.AllowedTenants, userInfo.Username)
 	if err != nil {
 		return nil, errors.FromError(err).ExtendComponent(searchEventStreamsComponent)
 	}
 
 	uc.logger.WithContext(ctx).Debug("event streams found successfully")
```
Comment (julien-marchand) [inline src/api/business/use-cases/event_streams/search.go:34]: Can we add event stream uuid and name here?

File: src/api/business/use-cases/event_streams/update.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	if err != nil {
+		return nil, errors.FromError(err).ExtendComponent(updateEventStreamComponent)
+	}
+
+	e, err := uc.db.FindOneByUUID(ctx, eventStream.UUID, userInfo.AllowedTenants, userInfo.Username)
+	if err != nil {
+		return nil, errors.FromError(err).ExtendComponent(updateEventStreamComponent)
+	}
+
+	logger.WithField("name", e.Name).Info("event stream updated successfully")
```
Comment (julien-marchand) [inline src/api/business/use-cases/event_streams/update.go:74]: Can we add event stream uuid here?

File: src/api/service/types/event_stream.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+import (
+	"time"
+
+	"github.com/consensys/orchestrate/src/entities"
+)
+
+type CreateWebhookEventStreamRequest struct {
+	Name    string            `json:"name" validate:"required" example:"my-webhook-stream"`
+	Chain   string            `json:"chain,omitempty" validate:"omitempty" example:"mainnet"`
+	URL     string            `json:"url" validate:"required" example:"https://my-event-steam-endpoint.com"`
```
Comment (julien-marchand) [inline src/api/service/types/event_stream.go:12]: Can we validate it's a proper URL?https://github.com/go-playground/validator#network

File: src/api/service/types/event_stream.go [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+	Headers map[string]string `json:"headers,omitempty" validate:"omitempty"`
+	Labels  map[string]string `json:"labels,omitempty" validate:"omitempty"`
+}
+
+func (r *CreateWebhookEventStreamRequest) ToEntity() *entities.EventStream {
+	return &entities.EventStream{
+		Name: r.Name,
+		Specs: &entities.Webhook{
+			URL:     r.URL,
```
Comment (julien-marchand) [inline src/api/service/types/event_stream.go:21]: same here

File: src/api/service/types/event_stream.go [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+		Channel: entities.EventStreamChannelWebhook,
+		Status:  entities.EventStreamStatusLive,
+		Labels:  r.Labels,
+	}
+}
+
+type CreateKafkaEventStreamRequest struct {
+	Name   string            `json:"name" validate:"required" example:"my-kafka-stream"`
+	Chain  string            `json:"chain,omitempty" validate:"omitempty" example:"mainnet"`
+	Topic  string            `json:"topic" validate:"required" example:"my-notification-topic"`
```
Comment (julien-marchand) [inline src/api/service/types/event_stream.go:33]: Can we add hostname & authentication parameters?

File: src/api/service/types/event_stream.go [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+		}
+	}
+
+	return es
+}
+
+type UpdateWebhookEventStreamRequest struct {
+	Name    string            `json:"name,omitempty" validate:"omitempty" example:"my-kafka-stream"`
+	Chain   string            `json:"chain,omitempty" validate:"omitempty" example:"mainnet"`
+	URL     string            `json:"url,omitempty" validate:"omitempty" example:"https://my-event-steam-endpoint.com"`
```
Comment (julien-marchand) [inline src/api/service/types/event_stream.go:78]: same here

File: src/api/service/types/event_stream.go [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+		Channel: entities.EventStreamChannelKafka,
+		Status:  entities.EventStreamStatusLive,
+		Labels:  r.Labels,
+	}
+}
+
+type UpdateKafkaEventStreamRequest struct {
+	Name   string            `json:"name,omitempty" validate:"omitempty" example:"my-kafka-stream"`
+	Chain  string            `json:"chain,omitempty" validate:"omitempty" example:"mainnet"`
+	Topic  string            `json:"topic,omitempty" validate:"omitempty" example:"my-notification-topic"`
```
Comment (julien-marchand) [inline src/api/service/types/event_stream.go:52]: Can we add hostname & authentication parameters?

**Review by julien-marchand — APPROVED** [review]

File: src/api/service/types/event_stream.go (hunk 10 above)
Comment (darioAnongba) [reply src/api/service/types/event_stream.go:52]: Already discussed with @ggarri , we cannot connect to an external dependency not controlled by Orchestrate:
- We can send notifications to the same Kafka as the internal one which is our dependency, same behavior as we have today. User defines the topic he wants in the event stream
- Connecting to an external Kafka is impossible unless we add a management API like we have for chains (or vaults in the QKM). This doesn't make sense for event streams.

File: src/api/service/types/event_stream.go (hunk 8 above)
Comment (darioAnongba) [reply src/api/service/types/event_stream.go:33]: Already discussed with @ggarri , we cannot connect to an external dependency not controlled by Orchestrate:
- We can send notifications to the same Kafka as the internal one which is our dependency, same behavior as we have today. User defines the topic he wants in the event stream
- Connecting to an external Kafka is impossible unless we add a management API like we have for chains (or vaults in the QKM). This doesn't make sense for event streams.

File: src/api/business/use-cases/event_streams/delete.go (hunk 1 above)
Comment (darioAnongba) [reply src/api/business/use-cases/event_streams/delete.go:42]: already in the ctx

File: src/api/business/use-cases/event_streams/get.go (hunk 2 above)
Comment (darioAnongba) [reply src/api/business/use-cases/event_streams/get.go:37]: already in ctx

File: src/api/business/use-cases/event_streams/create.go (hunk 3 above)
Comment (darioAnongba) [reply src/api/business/use-cases/event_streams/create.go:69]: uuid added

File: src/api/business/use-cases/event_streams/search.go (hunk 4 above)
Comment (darioAnongba) [reply src/api/business/use-cases/event_streams/search.go:34]: returns an array

File: src/api/business/use-cases/event_streams/update.go (hunk 5 above)
Comment (darioAnongba) [reply src/api/business/use-cases/event_streams/update.go:74]: added

File: src/api/service/types/event_stream.go (hunk 9 above)
Comment (darioAnongba) [reply src/api/service/types/event_stream.go:78]: added

File: src/api/service/types/event_stream.go (hunk 6 above)
Comment (darioAnongba) [reply src/api/service/types/event_stream.go:12]: added

**Pull request merged**